dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,329 @@
+                if(action.name().equalsIgnoreCase(actionStr)) {
+                    return action;
+                }
+                
+            }
+            return ACTIVATE;
+            
+            
+        }
+        
+        
+    }
+    
+    
+
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/{indexName: .*}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response deleteIndex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @PathParam("indexName") final String indexName) {
+
+        validateUser(request, response);
+        return Response.ok(new ResponseEntityView(APILocator.getContentletIndexAPI().delete(indexName))).build();
+
+
+    }
+
+
+    @PUT
+    @JSONP
+    @NoCache
+    @Path("/{indexName: .*}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response deactivateIndex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @PathParam("indexName") final String indexName, @QueryParam("action") final String action) throws DotDataException, IOException {
+
+        validateUser(request, response);
+        final IndexAction indexAction = IndexAction.fromString(action);
+        switch(indexAction){
+            case DEACTIVATE:
+                APILocator.getContentletIndexAPI().deactivateIndex(indexName);
+                break;
+            case CLEAR:
+                APILocator.getESIndexAPI().clearIndex(indexName);
+                break;
+            case OPEN:
+                APILocator.getESIndexAPI().openIndex(indexName);
+                break;
+            case CLOSE:
+                APILocator.getESIndexAPI().closeIndex(indexName);
+                break;
+            default:
+                APILocator.getContentletIndexAPI().activateIndex(indexName);
+            
+        }
+
+        
+        
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+
+
+
+    }
+    
+    
+    
+    
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response getReindexationProgress(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
+
+        validateUser(request, response);
+
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+
+    }
+
+    private final String DOTALL="DOTALL";
+    
+    
+    
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response startReindex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @QueryParam("shards") int shards, @DefaultValue(DOTALL) @QueryParam("contentType") String contentType) throws DotDataException, DotSecurityException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'contentType' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)